Feature rack group missing fields

jifox opened the pull request:
Add the following missing fields to module `rack_group`:

- region
- parent
- description

jifox commented:
@jmcgill298 @jvanderaa @FragmentedPacket

Is there anything missing that I should provide for merging into main branch?

FragmentedPacket commented:
@jifox Yes. I don't think this will properly resolve the parent rack group. Follow the `parent_region` [here]https://github.com/nautobot/nautobot-(ansible/blob/71c8dba5314d428c65f1022a2dcc64ce05cc7471/plugins/module_utils/utils.py#L391).

You'll need to copy most of what it does for that by adding `parent_rack_group` to convert back to just `parent`, but then `parent_rack_group` within `ALLOWED_QUERY_PARAMS`, etc.

jifox commented:
@FragmentedPacket Thanks for feedback, I'll try to sort this out

jifox commented:
@FragmentedPacket 

I've implemented as you suggested and used following test playbook to verify the functionality:

```yaml
- name: "Test Nautobot modules"
  connection: local
  hosts: localhost
  gather_facts: False

  vars:
    nburl: "http://localhost:8000"
    nbtoken: "dsSe5P656Z78x2fDsaGDK676shgf67ehfdhgJSAg"

  tasks:
    - name: Create site within Nautobot with only required information
      networktocode.nautobot.site:
        url: "{{ nburl }}"
        token: "{{ nbtoken }}"
        data:
          name: Site 1
          status: Active
        state: present

    - name: Create Parent Rack Group
      networktocode.nautobot.rack_group:
        url: "{{ nburl }}"
        token: "{{ nbtoken }}"
        data:
          name: Parent Rack Group 1
          site: Site 1
        state: present


    - name: Update description of parent rack group
      networktocode.nautobot.rack_group:
        url: "{{ nburl }}"
        token: "{{ nbtoken }}"
        data:
          name: Parent Rack Group 1
          site: Site 1
          description: Parent Rack Group Description
        state: present

    - name: Create member rack group
      networktocode.nautobot.rack_group:
        url: "{{ nburl }}"
        token: "{{ nbtoken }}"
        data:
          name: Member Rack Group 1
          site: Site 1
          parent_rack_group: Parent Rack Group 1
          description: "Member of 'Parent Rack Group 1'"
        state: present

    - name: Create member rack group 2
      networktocode.nautobot.rack_group:
        url: "{{ nburl }}"
        token: "{{ nbtoken }}"
        data:
          name: Member Rack Group 2
          site: Site 1
          parent_rack_group: Member Rack Group 1
          description: "Member of 'Member Rack Group 1'"
        state: present

    - name: Create member rack group 3
      networktocode.nautobot.rack_group:
        url: "{{ nburl }}"
        token: "{{ nbtoken }}"
        data:
          name: Member Rack Group 3
          site: Site 1
          parent_rack_group: Member Rack Group 1
          description: "Member of 'Member Rack Group 1'"
        state: present

    - name: Delete rack group within nautobot
      networktocode.nautobot.rack_group:
        url: "{{ nburl }}"
        token: "{{ nbtoken }}"
        data:
          name: Member Rack Group 3
          site: Site 1
        state: absent
```


```bash
$ ansible-playbook rack-group-test.yml
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Test Nautobot modules] **************************************************************************************************

TASK [Create site within Nautobot with only required information] *************************************************************
ok: [localhost]

TASK [Create Parent Rack Group] ***********************************************************************************************
ok: [localhost]

TASK [Update description of parent rack group] ********************************************************************************
ok: [localhost]

TASK [Create member rack group] ***********************************************************************************************
ok: [localhost]

TASK [Create member rack group 2] *********************************************************************************************
ok: [localhost]

TASK [Create member rack group 3] *********************************************************************************************
changed: [localhost]

TASK [Delete rack group within nautobot] **************************************************************************************
changed: [localhost]

PLAY RECAP ********************************************************************************************************************
localhost                  : ok=7    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 
```


![Screenshot 2021-04-11 074214](https://user-images.githubusercontent.com/8406301/114293826-b27ad400-9a99-11eb-8a23-6a7c2a7d31b3.jpg)

FragmentedPacket commented:
@jifox Thanks for your contributions!
